src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs

```diff
@@ -37,6 +39,11 @@ public class ReplicationDocumentSender : IDisposable
         private OutgoingReplicationStatsScope _statsInstance;
         internal readonly ReplicationStats _stats = new ReplicationStats();
         public bool MissingAttachmentsInLastBatch { get; private set; }
+        private HashSet<Slice> _deduplicatedAttachmentHashes = new(SliceComparer.Instance);
+        private Queue<Slice> _deduplicatedAttachmentHashesLru = new();
```

Comment:
The issue is that we might resend the same stream for different batches? since for the same batch the `_replicaAttachmentStreams` should handle it, right?

Reply:
note that we also read the entire stream from the disk. we might want to avoid this as well?

Reply:
The issue we have right now is network traffic, that is what I'm trying to optimize here.
If we already read the stream, it is already in memory, no additional cost.

And the issue is that if you have a high number of duplicated attachments, you are sending them (yes, in different batches). The scenario here is that we are sending ~500 - 1000 items per batch, but most of them are duplicate across batches.